torch/distributed/elastic/utils/store.py

```diff
@@ -109,7 +157,13 @@ def _barrier_nonblocking(store, world_size: int, key_prefix: str) -> str:
 
 
 def barrier(
-    store, world_size: int, key_prefix: str, barrier_timeout: float = 300
+    store,
+    world_size: int,
+    key_prefix: str,
+    barrier_timeout: float = 300,
+    rank: Optional[int] = None,
```

Comment:
Should we try to make this more general? For example, use this as a caller ID that could be anything since anyone could be using the barrier (even supposedly multiple processes on the same rank)?

Reply:
I was leaning towards just using identity vs rank, however I don't see other uses besides rank and it is familiar/consistent with other operations